### PR TITLE
feat: (IAC-1336) Add support for specifying K8s support plan

### PR DESCRIPTION
**Pull request by riragh**
## Changes:
This PR enables user to specify the K8s support plan for the AKS cluster to add support for AKS LTS K8s versions. New variable ~`aks_support_plan`~ `cluster_support_tier` is introduced with possible values `KubernetesOfficial` and `AKSLongTermSupport`. Defaults to `KubernetesOfficial`. 

AKSLongTermSupport can only be enabled along with SKU tier Premium. For details see [Long term Support](https://learn.microsoft.com/en-us/azure/aks/long-term-support) and for which K8s version has long term support see [AKS Kubernetes release calendar](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar).

## Tests:
Verified following scenarios, see internal ticket for details:
| Scenario | Provider | Task          | kubectl version | cadence   | Notes     |
|----------|----------|---------------|-----------------|-----------|-----------|
| 1        | Azure      | aks_cluster_sku_tier = "Premium", ~aks_support_plan~ cluster_support_tier = "AKSLongTermSupport",  network defaults: `aks_network_plugin = "kubenet", aks_network_policy = null, aks_network_plugin_mode = null ` | 1.27.9      |fast:2020 (2024.02) | |
| 2        | Azure      | aks_cluster_sku_tier = "Premium", ~aks_support_plan~ cluster_support_tier= "AKSLongTermSupport", aks_network_plugin = "azure", aks_network_policy = "azure", aks_network_plugin_mode = "overlay" | 1.27.9 | fast:2020 (2024.02) | |
| 3        | Azure      | aks_cluster_sku_tier = "Standard", ~aks_support_plan~ cluster_support_tier= "AKSLongTermSupport", network defaults | 1.27.9 | fast:2020 (2024.02) | This was negative test, received expected failure. LTS is only supported on SKU tier Premium. |
| 4        | Azure      | aks_cluster_sku_tier = "Standard", ~aks_support_plan~ cluster_support_tier= "KubernetesOfficial", network defaults  | 1.27.9| fast:2020 (2024.02) | |

### CODE REVIEW

**Review by dhoucgitter — APPROVED** [review]

File: docs/CONFIG-VARS.md [hunk 1]
```diff
@@ -194,7 +194,8 @@ Ubuntu 20.04 LTS is the operating system used on the Jump/NFS servers. Ubuntu cr
...
-| aks_cluster_sku_tier | Optimizes api server for cost vs availability | string | "Free" | Valid Values:  "Free", "Standard" | 
+| aks_cluster_sku_tier | The SKU Tier that should be used for this Kubernetes Cluster. Optimizes api server for cost vs availability | string | "Free" | Valid Values:  "Free", "Standard" and "Premium" | 
+| aks_support_plan | Specifies the support plan which should be used for this Kubernetes Cluster. Possible values are `KubernetesOfficial` and `AKSLongTermSupport`. | string | "KubernetesOfficial" | To enable long term K8s support is a combination of setting `aks_cluster_sku_tier` to `Premium` tier and explicitly selecting the `aks_support_plan` as `AKSLongTermSupport`. For details see [Long term Support](https://learn.microsoft.com/en-us/azure/aks/long-term-support) and for which K8s version has long term support see [AKS Kubernetes release calendar](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar).|
```
Comment (thpang) [inline docs/CONFIG-VARS.md:198]: The name here needs to change. Should be something like `cluster_support_tier` This is generic and can be used here in AKS without issue. The problem is now with the current name it'll be Azure specific when in reality the other cloud providers could also have something similar and at that point this repo would need to be adjusted.

**Review by thpang — CHANGES_REQUESTED** [review]
Variable name change

File: docs/CONFIG-VARS.md (hunk 1 above)
Comment (riragh) [reply docs/CONFIG-VARS.md]: Updated the variable name from `aks_support_plan` to `cluster_support_tier`.

**Review by thpang — APPROVED** [review]
LGTM

File: docs/CONFIG-VARS.md [hunk 2]
```diff
@@ -194,7 +194,8 @@ Ubuntu 20.04 LTS is the operating system used on the Jump/NFS servers. Ubuntu cr
...
-| aks_cluster_sku_tier | Optimizes api server for cost vs availability | string | "Free" | Valid Values:  "Free", "Standard" | 
+| aks_cluster_sku_tier | The SKU Tier that should be used for this Kubernetes Cluster. Optimizes api server for cost vs availability | string | "Free" | Valid Values:  "Free", "Standard" and "Premium" | 
+| cluster_support_tier | Specifies the support plan which should be used for this Kubernetes Cluster. Possible values are `KubernetesOfficial` and `AKSLongTermSupport`. | string | "KubernetesOfficial" | To enable long term K8s support is a combination of setting `aks_cluster_sku_tier` to `Premium` tier and explicitly selecting the `cluster_support_tier` as `AKSLongTermSupport`. For details see [Long term Support](https://learn.microsoft.com/en-us/azure/aks/long-term-support) and for which K8s version has long term support see [AKS Kubernetes release calendar](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar).|
```
Comment (sayeun) [inline docs/CONFIG-VARS.md:198]: To follow the way we use Description and Notes, I would suggest to move "Possible values are KubernetesOfficial and AKSLongTermSupport" from the Description column to the Notes column.

Comment (riragh) [reply docs/CONFIG-VARS.md]: Updated as per the comment.

**Review by sayeun — APPROVED** [review]

**Pull request merged**